app/actions/remote/session.test.ts

```diff
@@ -491,3 +493,167 @@ describe('logout', () => {
         }
     });
 });
+
+describe('nativeEntraLogin', () => {
+    const serverDisplayName = 'Test Server';
+    const serverIdentifier = 'test-server-id';
+    const intuneScope = 'api://test-scope/.default';
+
+    const mockTokens = {
+        accessToken: 'mock-access-token',
+        idToken: 'mock-id-token',
+        identity: {
+            upn: 'test@example.com',
+            tid: 'tenant-id',
+            oid: 'object-id',
+        },
+    };
+
+    let IntuneManager: any;
+
+    beforeEach(() => {
+        // Get the mocked IntuneManager
+        IntuneManager = require('@managers/intune_manager').default;
+
+        // Reset all mocks
+        jest.clearAllMocks();
+
+        // Set up default implementations
+        IntuneManager.login.mockResolvedValue(mockTokens);
+        IntuneManager.enrollServer.mockResolvedValue(undefined);
+        IntuneManager.isManagedServer.mockResolvedValue(false);
+
+        mockClient.loginByIntune.mockImplementation(() => Promise.resolve(user1));
+        mockGetCSRFFromCookie.mockImplementation(() => Promise.resolve('csrfid'));
+    });
+
+    it('should successfully login on first try without enrollment', async () => {
+        IntuneManager.isManagedServer.mockResolvedValue(false);
+
+        const result = await nativeEntraLogin(serverUrl, serverDisplayName, serverIdentifier, intuneScope);
+
+        expect(result.failed).toBe(false);
+        expect(IntuneManager.login).toHaveBeenCalledWith(serverUrl, [intuneScope]);
+        expect(mockClient.loginByIntune).toHaveBeenCalledWith(mockTokens.accessToken, expect.any(String));
+        expect(mockClient.setCSRFToken).toHaveBeenCalledWith('csrfid');
+        expect(IntuneManager.enrollServer).toHaveBeenCalledWith(serverUrl, mockTokens.identity);
+    });
+
```

Comment:
should there be a test for the 90 error when there is already a request for a silent refresh?